core/src/main/java/org/apache/iceberg/PositionDeletesTable.java

```diff
@@ -141,25 +147,88 @@ private Schema calculateSchema() {
     }
   }
 
+  /**
+   * Handle collisions between table and partition field ids, as both need to be part of position
+   * deletes table
+   *
+   * @param tableSchema original table schema
+   * @param partitionType original table's partition type
+   * @return partition type with reassigned field ids
+   */
+  public static Types.StructType partitionType(Schema tableSchema, Types.StructType partitionType) {
```

Comment:
These and following are just helper methods to reasisgn partition field ids to prevent collision with schema's field ids.